### PR TITLE
feat: beatutify busTooltip & make lineProfilePage actually work

### LINKED ISSUE #543: Visualize data about a bus-line
# Where is it?
1. go here:
image
2. go back in time :timer_clock: 
image
3. click a bus:
image
4. click the line title:
image

# The problem
It's an ugly, unfinished page

# What could you do?
Help us!
take the data (yes, the JSON thing) and make it look like something that a real user can actually read and understand

### PR DESCRIPTION
**Pull request by itsoriki**
Resolves #543 

# Description
- make LineProfilePage **fully functional**
- style `BusTooltip` according to language (also add a collapsible section)

## screenshots
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by itsoriki** [comment]
https://github.com/all-contributors please add @itsoriki as code contributor

**Comment by itsoriki** [comment]
@NoamGaash I can't figure out why that one job is failing consistently - No real error is printed to the console...
image

**Comment by NoamGaash** [comment]
@itsoriki yup that's bad
let's start with #747

**Comment by NoamGaash** [comment]
run `npm run lint` on your branch

**Comment by itsoriki** [comment]
@NoamGaash Sorry for the "mess" - can you review this PR now that everything has passed?

File: src/pages/components/map-related/MapLayers/BusToolTip.tsx [hunk 1]
```diff
@@ -53,17 +57,20 @@ export function BusToolTip({ position, icon }: BusToolTipProps) {
...
             </li>
             <li>
-              {t('velocity')} :<span>{`${position.point?.velocity}  ${t('kmh')}`}</span>
+              {`${t('velocity')}: `}
+              <span>{`${position.point?.velocity}  ${t('kmh')}`}</span>
             </li>
 
             <li>
-              {t('sample_time')} :
+              {`${t('sample_time')}: `}
```
Comment (NoamGaash) [inline src/pages/components/map-related/MapLayers/BusToolTip.tsx:73]: why not
```suggestion
              {t('sample_time')}:
```
?

File: src/pages/lineProfile/LineProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    search: { timestamp },
+    setSearch,
+  } = useContext(SearchContext)
+  const route = useLoaderData() as Route & { message?: string }
+  const [availableRoutes, setAvailableRoutes] = useState<BusRoute[] | null>(null)
+  const [selectedRouteKey, setSelectedRouteKey] = useState<string>('')
+
+  useEffect(() => {
+    window.scrollTo(0, 0)
+  }, [])
```
Comment (NoamGaash) [inline src/pages/lineProfile/LineProfile.tsx:42]: why is it needed?

File: src/pages/lineProfile/LineProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      </LineProfileMapContainer>
+    </div>
+  )
+}
+
+const LineProfileMapContainer = styled.div`
+  .map-info {
+    height: 15rem;
+  }
+`
```
Comment (NoamGaash) [inline src/pages/lineProfile/LineProfile.tsx:130]: should we move it to the scss file?

**Review by NoamGaash — APPROVED** [review]
That's a great improvement!
few suggestions:
1. I'm not sure that we need the expanding button - what is it good for?
2. We can consider translating the direction (290 מעלות) to geographic terms (צפון מערב, אזימוט 290)
3. when the tooltip is too tall. we can add an inner scrollbar or something similar

wdyt?

File: src/pages/components/map-related/MapLayers/BusToolTip.tsx (hunk 1 above)
Comment (itsoriki) [reply src/pages/components/map-related/MapLayers/BusToolTip.tsx:73]: The reason being the single space at the end of the string - doing it like you suggest will leave no space between the text and the data (unless we'll add an ugly `{' '}`) so I figured this is the cleanest approach to achieve the goal here

File: src/pages/lineProfile/LineProfile.tsx (hunk 2 above)
Comment (itsoriki) [reply src/pages/lineProfile/LineProfile.tsx:42]: I don't know why, but when you first arrive on this page, you are not at the top of it (i.e. you don't see the title for example) - see picture

image

Is there a better way to do it?

File: src/pages/lineProfile/LineProfile.tsx (hunk 3 above)
Comment (itsoriki) [reply src/pages/lineProfile/LineProfile.tsx:130]: Done

**Comment by itsoriki** [comment]
> That's a great improvement! few suggestions:
> 
> 1. I'm not sure that we need the expanding button - what is it good for?
> 2. We can consider translating the direction (290 מעלות) to geographic terms (צפון מערב, אזימוט 290)
> 3. when the tooltip is too tall. we can add an inner scrollbar or something similar
> 
> wdyt?

1. Removed the expanding button
2. I'll open a new (good-first) issue for that
3. Added an inner scrollbar to the `BusTooltip`

File: src/pages/lineProfile/LineProfile.tsx (hunk 2 above)
Comment (NoamGaash) [reply src/pages/lineProfile/LineProfile.tsx:41]: I'm not sure, we can investigate that later

**Comment by NoamGaash** [comment]
Thank you! :clap: 

@all-contributor please add @itsoriki for code :)

**Comment by NoamGaash** [comment]
maybe
@all-contributors please add @itsoriki for code 
?

**Pull request merged**